### PR TITLE
Add algebraic expressions support

### PR DESCRIPTION
**Pull request by gabrieldonadel**
## Description

- Add algebraic expressions support
- Add how to run instructions to README
- Update python files to support `pir` language
- Add `input.pir` file for lexical analysis

### What's still pending

- Take into consideration `\*, /, +, -` precedence order
- Add the ability to add parentheses in order to set the precedence order

### CODE REVIEW

File: pir/__main__.py [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
 try:
-    from .parser.cmmLexer import cmmLexer
-    from .parser.cmmParser import cmmParser
+    from .parser.pirLexer import pirLexer
+    from .parser.pirParser import pirParser
 except ImportError:
     # Generate ANTLR parser from g4 file
     from .generate_parser import generate_parser
-    generate_parser(path.join(path.dirname(__file__), 'parser/cmm.g4'))
+    generate_parser(path.join(path.dirname(__file__), 'parser/pir.g4'))
```
Comment (peterkrauz) [inline pir/__main__.py:12]: 💯

File: pir/parser/pir.g4 [hunk 2]
```diff
@@ -106,9 +131,60 @@ elseExpression: 'else' '{' generalDeclaration* '}';
...
+		| Identifier
+		| algebraicExpression
+	);
 comparisonOperator: '>' | '<' | '>=' | '<=' | '==' | '!=';
 
+algebraicExpression:
+	integerAlgebraicExpression
+	| floatAlgebraicExpression;
+
+// Interger
```
Comment (peterkrauz) [inline pir/parser/pir.g4]: extra `r`

File: pir/parser/pir.g4 [hunk 3]
```diff
@@ -106,9 +131,60 @@ elseExpression: 'else' '{' generalDeclaration* '}';
...
+		IntegerLiteral
+		| FloatLiteral
+		| Identifier
+		| algebraicExpression
+	);
 comparisonOperator: '>' | '<' | '>=' | '<=' | '==' | '!=';
 
+algebraicExpression:
+	integerAlgebraicExpression
+	| floatAlgebraicExpression;
```
Comment (peterkrauz) [inline pir/parser/pir.g4:144]: 👍

**Review by peterkrauz — APPROVED** [review]

File: pir/parser/pir.g4 (hunk 2 above)
Comment (gabrieldonadel) [reply pir/parser/pir.g4]: fixed

**Pull request merged**